### ISSUE TITLE
Can't use PrivateKeyFile on AWS without providing a KeyPairName (take 2)

### ISSUE THREAD

**Issue by ColinHebert** [mitchellh/packer#2301]
Follow up to #2281

It seems that it is still not possible to use aws without providing a keypair name:

> 1435111252,,ui,error,Build 'My CustomBuild' errored: Error launching source instance: InvalidKeyPair.NotFound: The key pair '' does not exist\n        status code: 400%!(PACKER_COMMA) request id: []

EDIT: a issue doesn't close another issue.
~~Closes~~ #2754

**Comment by catsby** [comment]
Hey @ColinHebert – can you give me an example of how you're specifying keys in the configuration? You're specifying both `ssh_keypair_name` and `ssh_private_key_file`, and your source AMI already has the key in place, correct?

**Comment by ColinHebert** [comment]
I'm only specifying the `ssh_private_key_file`, and not the `ssh_keypair_name` the key has already been set up on our base AMI, so it shouldn't be needed.

**Comment by ruizink** [comment]
Yep, I have the exactly same problem.
My pub key is already deployed, so neither `ssh_keypair_name` nor `temporary_key_pair_name` should be needed, since we are providing a `ssh_private_key_file`.

**Comment by kintoandar** [comment]
Same issue here, only using `ssh_private_key_file` as the AMI already has my public ssh key.

**Comment by ColinHebert** [comment]
Any news on this issue?

**Comment by MartinSGill** [comment]
Running into this issue as well. Anyone have a workaround, or know which release broke this, so I can downgrade to work around it please?

**Comment by ColinHebert** [comment]
@catsby Any news?

**Comment by ojongerius** [comment]
@catsby / @cbednarski is this waiting on something? Can we help to get this sorted?

**Comment by jkilada** [comment]
@catsby / @cbednarski Any word on this? Is there still no way to use an existing private key file without using a key pair name?

**Comment by heavyliftco** [comment]
Will this be fixed?

**Comment by oddtazz** [comment]
A fix for this issue would be nice. 
Had to pin packer to version 0.7.5 internally because of this bug.

**Comment by rickard-von-essen** [comment]
Please test if PR #3953 solves your issue.
